org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/XtextDocumentProvider.java

```diff
@@ -419,7 +428,15 @@ public boolean isSynchronized(Object element) {
 		} else {
 			return super.isSynchronized(element);
 		}
-		return synchronizationStamp == getModificationStamp(element);
+
+		// Check has to fit the following method: 
+		// org.eclipse.ui.editors.text.FileDocumentProvider.checkSynchronizationState(long, IResource)
+		boolean isSynchronized = synchronizationStamp == getSynchronizationStamp(input.getFile());
```

Comment:
I think I don't understand the idea behind this change.
Line 425 takes the modificationStamp and stores it as a local `synchronizationStamp`. Now we try to read `getSynchronizationStamp(someIFile)` which does
```
if (element instanceof IFileEditorInput) {
	FileInfo info= (FileInfo) getElementInfo(element);
	if (info != null)
		return info.fModificationStamp;
}
return super.getSynchronizationStamp(element);
```

it'll always go into the super.getSynchronizationStamp() which returns 0.
Something doesn't really work for me with this code.
Did you mean to call `computeModificationStamp` instead of getSynchronizationStamp.

If so: There must be something wrong with the tests. They should have caught this.
If not: Please explain (possibly again) in full detail what the idea is.

Reply:
You're right.. It could never be synchron, the file was reloaded usually. I guess I've to to store the time when the file was loaded and reloaded and compare with isSynchron.. Otherwise it will just compare the real files modification time stamp and the IResource modification timestamp.. But in this case the editor content has to be synchron, not just the file. 

I will add a few more asserts to check for isSynchronized() true/false, then we should be more secure with the tests.

Reply:
I've checked why this override was introduced, but then it doesn't makes a sense to keep this override. 
It was introduced to fix the the saving of files which are out of sync: https://bugs.eclipse.org/bugs/show_bug.cgi?id=343894. So If i remove this override, we will have the same problem back, but if I keep my changes in the XtextEditor class and call performSave(true, progressMonitor) .. (example from TextEditor which also forces the save) 

So might be someone has fixed the first bug in a wrong place and caused the unsync content problem.

Reply:
Removed the isSynchronized() overriden method, adjusted the performSave to force override .. Added more assertions to test.